docs/gitbook/architecture/topology/patterns.adoc

```diff
@@ -0,0 +1,164 @@
+providing the persistent association of the hospital patients with their
+"anonymous" insurance identifiers.
+
+
+== Combinations
+
+=== The Multi-Tier Hub and Spoke Molecule
+
+image:img/hierarchical_throughput.svg["The Multi-tier Hub and Spoke
+Model", 175, 194, role="right"]
+Using the Hub and Spoke molecule may present a performance problem if
+one or more spokes needs to dispatch more actions than the hub can
+handle. For example, in a blockchain network for a banking system,
+perhaps one branch in a bank is in the dense city center and processes
+triple the transactions of the other branches.
+
+The Multi-Tier Hub and Spoke molecule would be a good solution:
+the busy branch could be split into its own hub and spoke molecule,
+such as one blockchain for the branch itself and separate blockchains
+for each automated teller machine (ATM). The branch could then aggregate
+its data upwards to the hub at regular intervals. The splitting into
+multiple blockchains distributes the processing load -- provided that
+the new blockchains are hosted on separate nodes from the original node.
+
+
+In this model, a single layer of hub and spoke may not supply enough
+throughput to the system.
+
+Perhaps one branch in a bank is in the dense
+city center and processes triple the transactions of the other branches.
+This branch can then split into their own hierarchy, such as the branch
+itself and separate blockchains for each automated teller machine (ATM).
+They could then aggregate their data upwards, becoming their own spoke
+and hub molecule.
+
+This allows them to gain another potential order of magnitude more
+throughput without affecting the other branches.
+
+
+=== The Spokes as Control Model
+
+image:img/spoke_control.svg["The Spokes as Control Model", 175, 235,
+role="right"]
+In this model, each company has a chain that holds the entirety of
+their data. Much of that data is private, and so they do not permit
+direct connections between companies. However, they do share common
+goals and wish to work together.
+
+They form a consortium which manages a directory blockchain (the hub, in
+blue) which contains the hashes of the company chains participating in
+the consortium, and also maintains data integrity and trust system-wide.
+
+Whenever a company wishes to do business with another, they create and
+control an intermediary blockchain between them that contains all of
+the data required to conduct that business. None of the other parties
+are aware, maintaining privacy, and the structure of the intermediary
+blockchains, which could be atoms or other molecules, provides the
+processing required to perform all shared business operations.
+
+To satisfy compliance regulations, each company would share the hashes
+from their intermediary blockchains with the directory blockchain, to
+provide a verifiable timeline of their transactions without disclosing
+any details. An auditor could use this information to verify their
+compliance.
```

Comment:
Because these hashes are shared at regular intervals and not at every transaction, no metadata leakage occurs.